server/src/test/java/org/apache/iotdb/db/query/reader/series/SeriesAggregateReaderTest.java

```diff
@@ -0,0 +1,119 @@
+
+import static org.apache.iotdb.db.conf.IoTDBConstant.PATH_SEPARATOR;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.iotdb.db.engine.querycontext.QueryDataSource;
+import org.apache.iotdb.db.engine.storagegroup.TsFileResource;
+import org.apache.iotdb.db.exception.StorageEngineException;
+import org.apache.iotdb.db.exception.metadata.MetadataException;
+import org.apache.iotdb.db.exception.path.PathException;
+import org.apache.iotdb.db.exception.query.QueryProcessException;
+import org.apache.iotdb.db.query.aggregation.AggregateResult;
+import org.apache.iotdb.db.query.context.QueryContext;
+import org.apache.iotdb.db.query.factory.AggreResultFactory;
+import org.apache.iotdb.tsfile.exception.write.WriteProcessException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.write.schema.MeasurementSchema;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SeriesAggregateReaderTest {
+
+  private static final String SERIES_READER_TEST_SG = "root.seriesReaderTest";
+  private List<String> deviceIds = new ArrayList<>();
+  private List<MeasurementSchema> measurementSchemas = new ArrayList<>();
+
+  private List<TsFileResource> seqResources = new ArrayList<>();
+  private List<TsFileResource> unseqResources = new ArrayList<>();
+
+
+  @Before
+  public void setUp() throws MetadataException, PathException, IOException, WriteProcessException {
+    SeriesReaderTestUtil.setUp(measurementSchemas, deviceIds, seqResources, unseqResources);
+  }
+
+  @After
+  public void tearDown() throws IOException, StorageEngineException {
+    SeriesReaderTestUtil.tearDown(seqResources, unseqResources);
+  }
+
+  @Test
+  public void aggregateTest() {
+    try {
+      Path path = new Path(SERIES_READER_TEST_SG + PATH_SEPARATOR + "device0", "sensor0");
+      QueryDataSource queryDataSource = new QueryDataSource(path, seqResources, unseqResources);
+      SeriesAggregateReader seriesReader = new SeriesAggregateReader(path, TSDataType.INT32,
+          new QueryContext(), queryDataSource, null, null);
+      AggregateResult aggregateResult = AggreResultFactory
+          .getAggrResultByName("count", TSDataType.INT32);
+      while (seriesReader.hasNextChunk()) {
+        if (seriesReader.canUseCurrentChunkStatistics()) {
+          Statistics chunkStatistics = seriesReader.currentChunkStatistics();
+          aggregateResult.updateResultFromStatistics(chunkStatistics);
+          seriesReader.skipCurrentChunk();
+          continue;
+        }
+        int loopTime = 0;
+        while (seriesReader.hasNextPage()) {
+          if (seriesReader.canUseCurrentPageStatistics()) {
```

Comment:
Why is this statement always false here?

Reply:
Because there is only one page in a chunk of his test data, the non-overlapped pages will re-enter the chunk layer to judge once and then the are calculated directly using chunkMetaData.

Reply:
Then this test should be fixed.

Reply:
Sorry @LeiRui , it's my fault. I have fixed this test in the latest commit. Thanks for your further review!